### ISSUE TITLE
self.loss1 = tf.reduce_mean(self.reward_baseline * self.log_softmax, 0) - 1 * self.lr1 * tf.reduce_mean(self.entropy_regularization, 0)

### ISSUE THREAD

**Issue by Foyn** [huawei-noah/trustworthyAI#4]
Hello! I have read all of your code and the great paper published in ICLR2020.
I have some questions about the code:
1 . in train step1, why the loss1 of the actor has to be:
   _self.loss1 = tf.reduce_mean(self.reward_baseline * self.log_softmax, 0) - 1 * self.lr1 * 
  tf.reduce_mean(self.entropy_regularization, 0)_
  i didn't find any formula or explanation in the paper.
2. What does **reward base line** in this code use for?
I would be appreciated if you can answer these questions.

**Comment by zhushy** [comment]
1. this corrsponds to the expected reward J given in Eq. (7). For each graph, we can calculate its probablility and the rest follows from policy gradient algorithm.  The second term is an entropy regularization term to encourage exploitation for RL agent.

2. This is used for reducing the variance of the policy graident. See lines 137-138:
`self.reward_baseline= tf.stop_gradient(self.reward - self.avg_baseline - self.critic.predictions)`

**Comment by Foyn** [comment]
> 1. this corrsponds to the expected reward J given in Eq. (7). For each graph, we can calculate its probablility and the rest follows from policy gradient algorithm.  The second term is an entropy regularization term to encourage exploratio for RL agent.
> 2. This is used for reducing the variance of the policy graident. See lines 137-138:
>    `self.reward_baseline= tf.stop_gradient(self.reward - self.avg_baseline - self.critic.predictions)`

Thank you very much

**Comment by Foyn** [comment]
Thank you! Can I take the probablility of each graph as a weight  and the total batch reward as the weight sum of the reward of graphs, from the _weight_ perspective?
And I still have a question, in the code, the trainstep1 and trainstep2 share the self.opt1. 
_self.train_step2 = self.opt1.apply_gradients(capped_gvs2, global_step=self.global_step2)_
I think the critic, that is, the train step2 should use its own self.opt2

**Comment by zhushy** [comment]
Sorry, I did not check the repo for long. 
 
Our case is just one-step episodic. The \sum_{over batch}  (negative log-likelihood) * reward is treated as loss function. 

Yes, that should be self.opt2; this might need some tuning to make it work as before, and to keep results reproducible, we would like to leave it as was. Thanks very much for pointing this out.
